hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/HoodieMultiTableDeltaStreamer.java

```diff
@@ -378,16 +383,23 @@ private static String resetTarget(Config configuration, String database, String
   /**
    * Creates actual HoodieDeltaStreamer objects for every table/topic and does incremental sync.
    */
-  public void sync() {
-    for (TableExecutionContext context : tableExecutionContexts) {
-      try {
-        new HoodieDeltaStreamer(context.getConfig(), jssc, Option.ofNullable(context.getProperties())).sync();
-        successTables.add(Helpers.getTableWithDatabase(context));
-      } catch (Exception e) {
-        logger.error("error while running MultiTableDeltaStreamer for table: " + context.getTableName(), e);
-        failedTables.add(Helpers.getTableWithDatabase(context));
-      }
-    }
+  public void sync() throws InterruptedException {
+    ExecutorService executorService = Executors.newFixedThreadPool(tableExecutionContexts.size());
+    tableExecutionContexts.forEach(context -> {
+      executorService.execute(new Runnable() {
+        @Override
+        public void run() {
+          try {
+            new HoodieDeltaStreamer(context.getConfig(), jssc, Option.ofNullable(context.getProperties())).sync();
+            successTables.add(Helpers.getTableWithDatabase(context));
+          } catch (Exception e) {
+            logger.error("error while running MultiTableDeltaStreamer for table: " + context.getTableName(), e);
+            failedTables.add(Helpers.getTableWithDatabase(context));
+          }
+        }
+      });
+    });
+    executorService.shutdown();
```

Comment:
should we add awaitTermination here ? we can't proceed w/ next batch until all tables in current batch is completed right?